realm/src/androidTest/java/io/realm/RealmConfigurationTest.java

```diff
@@ -0,0 +1,255 @@
+        try {
+            Realm.getDefaultInstance();
+        } catch (NullPointerException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testGetNullInstance() {
+        try {
+            Realm.getInstance((RealmConfiguration) null);
+        } catch (NullPointerException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testNullDirThrows() {
+        try {
+            new RealmConfiguration.Builder((File) null).build();
+        } catch (IllegalArgumentException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testNullNameThrows() {
+        try {
+            new RealmConfiguration.Builder(getContext()).name(null).build();
+        } catch (IllegalArgumentException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testEmptyNameThrows() {
+        try {
+            new RealmConfiguration.Builder(getContext()).name("").build();
+        } catch (IllegalArgumentException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testNullKeyThrows() {
+        try {
+            new RealmConfiguration.Builder(getContext()).encryptionKey(null).build();
+        } catch (IllegalArgumentException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testWrongKeyLengthThrows() {
+        try {
+            new RealmConfiguration.Builder(getContext()).encryptionKey(new byte[63]).build();
+        } catch (IllegalArgumentException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testNegativeVersionThrows() {
+        try {
+            new RealmConfiguration.Builder(getContext()).schemaVersion(-1).build();
+        } catch (IllegalArgumentException expected) {
+            return;
+        }
+        fail();
+    }
+
+    public void testVersionLessThanDiscVersionThrows() {
+        realm = Realm.getInstance(new RealmConfiguration.Builder(getContext()).schemaVersion(42).build());
+        realm.close();
+
+        try {
+            Realm.getInstance(new RealmConfiguration.Builder(getContext()).schemaVersion(1).build());
+        } catch (IllegalArgumentException expected) {
+            return;
+        }
+        fail();
+    }
+
+    // TODO Should throw IllegalState instead
+    public void testVersionEqualWhenSchemaChangesThrows() {
+        realm = Realm.getInstance(new RealmConfiguration.Builder(getContext())
+                .deleteRealmBeforeOpening()
+                .schemaVersion(42)
+                .schema(Dog.class)
+                .create());
+        realm.close();
+
+        try {
+            Realm.getInstance(new RealmConfiguration.Builder(getContext())
+                    .schemaVersion(42)
+                    .schema(AllTypesPrimaryKey.class)
+                    .build());
+            fail("A migration should be required");
```

Comment:
It seems to be an inconsistency how you use `try`/`catch` and `fail()` in this unit test and other unit tests.